IdentityCore/src/cache/mac/MSIDMacKeychainTokenCache.m

```diff
@@ -364,6 +364,78 @@ + (MSIDMacKeychainTokenCache *)defaultKeychainCache
 
 #pragma mark - init
 
+- (BOOL)isAppEntitled
+{
+    static dispatch_once_t once;
+    static BOOL appEntitled = NO;
+    
+    dispatch_once(&once, ^{
+        SecCodeRef selfCode = NULL;
+        SecCodeCopySelf(kSecCSDefaultFlags, &selfCode);
+        
+        if (selfCode)
+        {
+            CFDictionaryRef cfDic = NULL;
+            SecCodeCopySigningInformation(selfCode, kSecCSSigningInformation, &cfDic);
+            
+            if (!cfDic)
+            {
+                MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"Failed to retrieve code signing information");
+            }
+            else
+            {
+                NSDictionary *signingDic = CFBridgingRelease(cfDic);
+                NSDictionary *entitlementsDictionary = [signingDic msidObjectForKey:(__bridge NSString*)kSecCodeInfoEntitlementsDict ofClass:[NSDictionary class]];
+                NSArray *keychainGroups = [entitlementsDictionary msidObjectForKey:@"keychain-access-groups" ofClass:[NSArray class]];
+                
+                for (id element in keychainGroups) {
+                    if ([element hasSuffix:@"com.microsoft.identity.universalstorage"])
+                    {
+                        appEntitled = YES;
+                        break;
+                    }
+                }
+            }
+            
+            CFRelease(selfCode);
+        }
+    });
+    
+    return appEntitled;
+}
+
+-(BOOL)shouldUseLoginKeychain
+{
+    if (@available(macOS 10.15, *))
+    {
+        
+        if ([[NSUserDefaults standardUserDefaults] boolForKey:kLoginKeychainEmptyKey])
+        {
+#if MS_INTERNAL_BUILD
+            if ([self isAppEntitled])
```

Comment:
nit: can be simplified to:
`return ![self isAppEntitled];`
instead of `if ... else ...`